From a14028b0878c1ba27f7c7a6d0962874d0f7f3801 Mon Sep 17 00:00:00 2001
From: Samuel Martin <s.martin49@gmail.com>
Date: Sun, 31 Aug 2014 11:54:37 +0200
Subject: [PATCH 3/3] cmake: use the standard CMake flag to drive the shared
 object build

If BUILD_SHARED_LIBS is set and not USE_SHARED_POLARSSL_LIBRARY, then
drive USE_SHARED_POLARSSL_LIBRARY with the BUILD_SHARED_LIBS value.

Signed-off-by: Samuel Martin <s.martin49@gmail.com>
---
 library/CMakeLists.txt | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/library/CMakeLists.txt b/library/CMakeLists.txt
index 27bd2e0..2ae0aba 100644
--- a/library/CMakeLists.txt
+++ b/library/CMakeLists.txt
@@ -1,5 +1,15 @@
-option(USE_STATIC_POLARSSL_LIBRARY "Build PolarSSL static library." ON)
-option(USE_SHARED_POLARSSL_LIBRARY "Build PolarSSL shared library." OFF)
+# Use the standard CMake flag to drive the shared object build.
+if(DEFINED BUILD_SHARED_LIBS AND NOT DEFINED USE_STATIC_POLARSSL_LIBRARY AND NOT DEFINED USE_SHARED_POLARSSL_LIBRARY)
+  set(USE_STATIC_POLARSSL_LIBRARY ON)
+  if(BUILD_SHARED_LIBS)
+    set(USE_SHARED_POLARSSL_LIBRARY ON)
+  else()
+    set(USE_SHARED_POLARSSL_LIBRARY OFF)
+  endif()
+else()
+  option(USE_STATIC_POLARSSL_LIBRARY "Build PolarSSL static library." ON)
+  option(USE_SHARED_POLARSSL_LIBRARY "Build PolarSSL shared library." OFF)
+endif()
 
 set(src
      aes.c
-- 
2.1.0

